app/services/customers/eu_auto_taxes_service.rb

```diff
@@ -32,7 +32,18 @@ def vies_check
 
     def process_vies_tax(customer_vies)
       if organization_country_code.casecmp?(customer_vies[:country_code])
```

Comment:
Hey!
since we're touching this method, what do you think about this slight refactoring of the whole method?
```
def process_vies_tax(customer_vies)
  return  'lago_eu_reverse_charge' unless organization_country_code.casecmp?(customer_vies[:country_code])

  standard_code = "lago_eu_#{organization_country_code.downcase}_standard"
  return standard_code if eu_country_exceptions(country_code: customer_vies[:country_code]).empty?
  return standard_code if customer.zipcode.blank?
  return standard_code if applicable_tax_exceptions(country_code: customer_vies[:country_code]).blank?

  exception_code = applicable_tax_exceptions(country_code: customer_vies[:country_code]).first['name'].parameterize.underscore
  "lago_eu_#{customer_vies[:country_code].downcase}_exception_#{exception_code}"
end

def applicable_tax_exceptions(country_code:)
  @applicable_tax_exceptions ||= eu_country_exceptions(country_code: country_code).select do |exception|
    customer.zipcode.match?(exception['postcode'])
  end
end

def eu_country_exceptions(country_code:)
  LagoEuVat::Rate.new.country_rates(country_code:)[:exceptions]
end
```

Reply:
good catch, I just did it!